durabletask/internal/grpc_interceptor.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from collections import namedtuple
+from typing import List, Tuple
+
+import grpc
+
+
+class _ClientCallDetails(
```

Comment:
I have used single implementation for all interceptors now.
Also, would need to create an implementation for grpc.ClientCallDetails interface and also create new `metadata` object.
As, client_call_details.metadata is otherwise not accessible in required way.
Also, client_call_details.metadata is read-only.

<img width="431" alt="image" src="https://github.com/microsoft/durabletask-python/assets/24491296/b00b3840-56e5-47c9-a801-a578dd2099ac">

Please refer: https://github.com/grpc/grpc/blob/62521a889fe134859aed7bdf4ea0730534f15d09/examples/python/interceptors/headers/header_manipulator_client_interceptor.py#L38

https://stackoverflow.com/a/54743669/18848251